lib/services/local/access_list.go

```diff
@@ -732,6 +733,9 @@ func (a *AccessListService) UpsertAccessListWithMembers(ctx context.Context, acc
 					if existingMember.Spec.Reason != "" {
 						newMember.Spec.Reason = existingMember.Spec.Reason
 					}
+					if existingMember.Origin() == common.OriginAWSIdentityCenter {
+						newMember.Metadata.Labels = existingMember.GetAllLabels()
+					}
```

Comment:
It might be helpful to add a comment here explaining the purpose of this code.

As I understand it, the frontend currently doesn’t support modifying user member labels. This code appears to preserve the existing label state from the previous object, as the `UpsertAccessListWithMembers` handling in the frontend leaves member labels empty.


Would it be possible to use newMember.AddedBy.AddedBy = "IGS Service" instead of labels? If so, it might allow us to avoid making this change altogether.  

My motivation is that this logic of label preservation/forwarding  is quite complex and technical dept so if possible i would suggest to avoid adding additional label forwarding logic.  


If we proceed with this change the UpdateAccessListMember endpoint should be also aligned: 
https://github.com/gravitational/teleport.e/blob/9017e6e8ab8ddea071a644cde785f1e7f637b6f6/lib/web/accesslist.go#L237

Reply:
Probably this `addMembersToAccessList ` endpoint was likely added and exposed to the frontend, but currently, it’s not in use and replaced by` UpsertAccessListWithMembers`.